src/auth/SignUp.jsx

```diff
@@ -9,48 +9,116 @@ import {
-          </div>
-          <div>
-            <Input placeholder='Nickname' width='100%' />
-          </div>
-          <div>
-            <Input placeholder='Password' width='100%' />
-          </div>
+const initialNewUserState = {
+  firstName: '',
+  lastName: '',
+  email: '',
+  nickName: '',
+  password: '',
+  repeatedPassword: '',
+};
+
+export const SignUp = () => {
+  const [newUser, setNewUser] = useState(initialNewUserState);
+  const { dispatchUser } = useUser();
+
+  const handleSignUpUser = (e) => {
+    e.preventDefault();
+    const { firstName, lastName, email, nickName, password } = newUser;
+    signUpUser({
+      firstName,
+      lastName,
+      email,
+      nickName,
+      password,
+      dispatchUser,
+    });
+    setNewUser(initialNewUserState);
+  };
+
+  const handleInputs = (e) =>
+    setNewUser({ ...newUser, [e.target.name]: e.target.value });
+
+  return (
+    <SignUpStyles>
+      <GridAuth height='800px'>
+        <FormStlyes>
           <div>
-            <Input placeholder='Repeat password' width='100%' />
+            <Title>
+              Hurrraaaaay! <br />
+              Lets us know who you are!
+            </Title>
+            <PrimaryText>
+              We won&apos;t share your info with anybody. I promise
+            </PrimaryText>
           </div>
-          <ButtonTertiaryStyles>
-            <Button secondary type='button'>
-              Create account
-            </Button>
-          </ButtonTertiaryStyles>
-        </Form>
-      </FormStlyes>
-      <ImageSignUp background={backgroundImage} />
-    </GridAuth>
-  </SignUpStyles>
-);
+          <Form onSubmit={handleSignUpUser}>
+            <FirstLastStyles>
+              <Input
+                name='firstName'
+                value={newUser.firstName}
+                onChange={handleInputs}
+                placeholder='First Name'
+              />
+              <Input
+                name='lastName'
+                value={newUser.lastName}
+                onChange={handleInputs}
+                placeholder='Last Name'
+              />
+            </FirstLastStyles>
+            <div>
+              <Input
+                name='email'
+                value={newUser.email}
+                onChange={handleInputs}
+                placeholder='Email'
+                width='100%'
+              />
+            </div>
+            <div>
+              <Input
+                name='nickName'
+                value={newUser.nickName}
+                onChange={handleInputs}
+                placeholder='Nickname'
+                width='100%'
+              />
+            </div>
+            <div>
+              <Input
+                name='password'
+                value={newUser.password}
+                onChange={handleInputs}
+                placeholder='Password'
+                width='100%'
+              />
+            </div>
+            <div>
+              <Input
```

Comment:
we could also  check if the repeated password the same is like the password

Reply:
Yes, we need validation for the form. But I thought it would be good to do it in another PR this one was already 185 lines

Reply:
Long PR are harder to review, it is more work for you :D

Reply:
i know i  am really realy sorry for the big one. it will not happen again